module-2/backend/week_two.md

```diff
@@ -5,16 +5,36 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+  - it is a bridge between your ruby code and the sql database
 2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+    class - they apply to every instance in that class
 3. What do they allow you to do?
+  allow you to define 1-m, 1-1, m-m, relationships between tables
 4. What's the difference between agile workflow and waterfall method?
+  agile - go step by step, heavy focus on refactoring at the end of each step and making sure that the endgame is still relevant/appropriate
+  waterfall - define objective at beginning, go headstrong through scheduled set of steps and hope that by the end of the project the endgame is still appropriate
 5. What is the difference between `#new` and `#create`?
+  new doesn't save to database, create does
 6. At a basic level, what does cURL allow you to do?
+  trigger a browser request from terminal
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+  many students to one teacher . . . a student only has one teacher, a teacher has many students
+    students          teachers
+    id                id
+    name              name
+    teacher_id
+    
 8. Define foreign key, primary key, and schema.
+  foreign_key - points from a many to the one
+  primary_key - unique identifier for each entry in a table, used in conjunction with foreign keys
+  schema - overall database structure
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+  they are the same number, the foreign matches whatever primary key it wants to relate to
 10. What are the parts of an HTTP response?
+  headers - verb, path, protocol, a few other things
```

Comment:
response is going to have your status (e.g. 200, 404, 503) in the headers. Verb is going to be in your request.